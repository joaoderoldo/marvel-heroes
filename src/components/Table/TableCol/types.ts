export interface TableColProps {
  children: React.ReactNode;
  classes?: string | string[];
  dataLabel?: string;
}
