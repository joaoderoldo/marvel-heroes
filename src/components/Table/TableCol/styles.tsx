import styled from "styled-components";

export const TableColWrapper = styled.div`
  width: calc(100% / 3);
  display: flex;
  align-items: center;

  &:last-child {
    justify-content: flex-end;
  }

  @media all and (max-width: 767px) {
    display: flex;
    padding: 10px 0;
    width: 100%;

    &:last-child {
      justify-content: flex-start;
    }

    &:before {
      color: #6c7a89;
      padding-right: 10px;
      content: attr(data-label);
      width: 40%;
      display: flex;
      flex-shrink: 0;
    }
  }
`;
