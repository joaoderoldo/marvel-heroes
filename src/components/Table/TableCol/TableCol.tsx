import clsx from "clsx";
import { FC } from "react";

import { TableColWrapper } from "./styles";
import { TableColProps } from "./types";

const TableCol: FC<TableColProps> = ({ children, classes, dataLabel }) => {
  return (
    <TableColWrapper className={clsx(classes)} data-label={dataLabel}>
      {children}
    </TableColWrapper>
  );
};

export default TableCol;
