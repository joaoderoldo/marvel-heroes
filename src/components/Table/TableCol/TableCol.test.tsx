import { render } from "@testing-library/react";
import TableCol from "./TableCol";

describe("TableCol", () => {
  it("should render the component", () => {
    const { container } = render(
      <TableCol dataLabel="Name" classes="table-cell">
        John Doe
      </TableCol>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render children correctly", () => {
    const { getByText } = render(
      <TableCol dataLabel="Name" classes="table-cell">
        John Doe
      </TableCol>
    );
    expect(getByText("John Doe")).toBeInTheDocument();
  });

  it("should render data-label attribute correctly", () => {
    const { getByText } = render(
      <TableCol dataLabel="Name" classes="table-cell">
        John Doe
      </TableCol>
    );
    expect(getByText("John Doe")).toHaveAttribute("data-label", "Name");
  });
});
