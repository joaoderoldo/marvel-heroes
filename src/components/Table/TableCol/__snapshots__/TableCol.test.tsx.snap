// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableCol > should render the component 1`] = `
<div
  class="sc-beqWaB kzVmMa table-cell"
  data-label="Name"
>
  John Doe
</div>
`;
